Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/MainAddActionsNavigationPage.xaml.cs

```diff
@@ -124,11 +133,37 @@ private void XNavActLib_Click(object sender, RoutedEventArgs e)
 
         private void XNavSpy_Click(object sender, RoutedEventArgs e)
         {
-            if (mLiveSpyNavPage == null)
+            LiveSpyPageLoaded = true;
+            LiveSpyNavPage liveSpyNavPage = GetLiveSpyPage();
+            LoadActionFrame(liveSpyNavPage, "Live Spy", eImageType.Spy);
+        }
+
+        /// <summary>
+        /// This method is used to get the new LiveSpyPage based on Context and Agent
+        /// </summary>
+        /// <returns></returns>
+        private LiveSpyNavPage GetLiveSpyPage()
+        {
+            LiveSpyNavPage liveSpyNavPage = null;
+            if (mLiveSpyNavPages != null && mLiveSpyNavPages.Count > 0)
```

Comment:
you can keep the unique spy page per Agent but please move all relevent code from main page to "LiveSpyNavPage" page, so you will have "LiveSpyPage" and "LiveSpyNavPage" which will manage the agent-spyPage dic and will show in Frame the "LiveSpyPage"

Reply:
Added new SpyPage